piecrust-uplink/src/types.rs

```diff
@@ -40,26 +43,34 @@ pub const MODULE_ID_BYTES: usize = 32;
 pub struct ModuleId([u8; MODULE_ID_BYTES]);
 
 impl ModuleId {
+    /// Creates a placeholder [`ModuleId`] until the host deploys the module
+    /// and sets a real [`ModuleId`]
```

Comment:
An uninitialized `ModuleId` can also be used to see if a module is the first to be called. Maybe put this in `is_uninitialized` and refer to it here?